### PR TITLE
switch to fs.promises vs bluebird for tests

### LINKED ISSUE #1791: Testing: Drop Bluebird, use utils.promisify
As promises are now native, and promisify() ships with Node’s utils, we can remove our dependency on Bluebird. 

We can also use async/await to make code in the test directory more maintainable.

### PR DESCRIPTION
**Pull request by yyyc514**
Potentially closes #1791

### CODE REVIEW

**Comment by yyyc514** [comment]
@marcoscaceres I whipped this up quickly, but I'm not really sure I see the advantage of using the async stuff here when in almost all the cases we're doing things sequentially in a single thread... wouldn't it be easier and simpler to just use the Sync methods?

**Comment by yyyc514** [comment]
Such as:

```
return fs.readdir(markupPath).then((list) => list.forEach(testLanguage));
// vs
return fs.readdirSync(markupPath).forEach(testLanguage);
```

File: test/detect/index.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 'use strict';
 
-let bluebird = require('bluebird');
-let fs       = bluebird.promisifyAll(require('fs'));
+let fs       = require('fs').promises
```
Comment (marcoscaceres) [inline test/detect/index.js]:
```suggestion
const fs = require('fs').promises
```

File: test/detect/index.js [hunk 2]
```diff
@@ -15,18 +14,19 @@ function testAutoDetection(language) {
   const languagePath = utility.buildPath('detect', language);
 
   it(`should have test for ${language}`, function() {
```
Comment (marcoscaceres) [inline test/detect/index.js]:
```suggestion
  it(`should have test for ${language}`, async () => {
```

File: test/detect/index.js [hunk 3]
```diff
@@ -15,18 +14,19 @@ function testAutoDetection(language) {
...
-    return fs.statAsync(languagePath)
+    return fs.stat(languagePath)
```
Comment (marcoscaceres) [inline test/detect/index.js]:
```suggestion
    const path = await fs.stat(languagePath);
```

File: test/detect/index.js [hunk 4]
```diff
@@ -15,18 +14,19 @@ function testAutoDetection(language) {
...
       .then(path => path.isDirectory().should.be.true);
```
Comment (marcoscaceres) [inline test/detect/index.js]:
```suggestion
      path.isDirectory().should.be.true;
```

File: test/detect/index.js [hunk 5]
```diff
@@ -15,18 +14,19 @@ function testAutoDetection(language) {
...
   });
 
-  it(`should be detected as ${language}`, function() {
-    return fs.readdirAsync(languagePath)
+  it(`should be detected as ${language}`, async function() {
+    let dirs = await fs.readdir(languagePath)
```
Comment (marcoscaceres) [inline test/detect/index.js]: Only use `let` for reassignment. 

```suggestion
    const dirs = await fs.readdir(languagePath)
```

File: test/detect/index.js [hunk 6]
```diff
@@ -15,18 +14,19 @@ function testAutoDetection(language) {
...
+    let files = await Promise.all(dirs
```
Comment (marcoscaceres) [inline test/detect/index.js]:
```suggestion
    const files = await Promise.all(dirs
```

File: test/special/index.js [hunk 7]
```diff
@@ -1,10 +1,9 @@
 'use strict';
 
 let _        = require('lodash');
-let bluebird = require('bluebird');
 let hljs     = require('../../build');
 let { JSDOM } = require('jsdom');
-let readFile = bluebird.promisify(require('fs').readFile);
+let readFile = require('fs').promises.readFile
```
Comment (marcoscaceres) [inline test/special/index.js]:
```suggestion
let { readFile } = require('fs').promises;
```

File: test/tools.js [hunk 8]
```diff
@@ -1,9 +1,8 @@
 'use strict';
 
 let utility = require('../tools/utility'),
-    bluebird = require('bluebird'),
     path = require('path'),
-    readFile = bluebird.promisify(require('fs').readFile);
+    readFile = require('fs').promises.readFile
```
Comment (marcoscaceres) [inline test/tools.js]:
```suggestion
    { readFile } = require('fs').promises;
```

File: test/detect/index.js (hunk 1 above)
Comment (yyyc514) [reply test/detect/index.js]: Sure if we're doing that I'll change them all.  I was doing the minimum work, not rewriting. :-)  But I can.

**Comment by yyyc514** [comment]
> Looks good. Is there more bluebird around the place?

Probably in build but I plan to redo all that, so not worth the time.

**Pull request merged**